### PR TITLE
refactor(api): project variables crypto

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12650.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✘

File: engine/api/project/dao_variable.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+	}
+
+	if sdk.NeedPlaceholder(v.Type) && v.Value == sdk.PasswordPlaceholder {
+		return fmt.Errorf("You try to insert a placeholder for new variable %s", v.Name)
+	}
+
+	dbVar := newDBProjectVariable(*v, projID)
+
+	err := gorpmapping.InsertAndSign(context.Background(), db, &dbVar)
+	if err != nil && strings.Contains(err.Error(), "project_variable_pkey") {
```
Comment (richardlt) [inline engine/api/project/dao_variable.go]: No err type for this purpose ?

Comment (fsamin) [reply engine/api/project/dao_variable.go]: you're right it was a weird existing err management

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12705.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12705.1 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Pull request merged**